This jupyter notebook contains code to answer some questions on the Court precedent Dataset used in my thesis.

In [1]:
import pandas as pd
df_court = pd.read_pickle('court_decisions.pkl')

How many court decision are in the dataset?

In [2]:
len(df_court)

249581

How many court decision include legal paragraphs(explicitly tagged)?

                                    _id Entscheidungsdatum  \
0  {'$oid': '65c6877f380bbdc9006d66d0'}         13.06.2020   

                                     Gericht Aktenzeichen  ECLI  \
0  Thüringer Oberverwaltungsgericht 3. Senat  3 EN 374/20  None   

                                        Entscheidung  \
0  Leitsatz\n\n\n1. Zu den rechtlichen Voraussetz...   

                                                Norm  Bundesland  \
0  Art 2 Abs 1 GG,§ 6 CoronaVGrundV TH, § 47 Abs ...  thueringen   

                                                Link original  
0  https://landesrecht.thueringen.de/bsth/MWRE200...      NaN  


In [9]:
df_court.iloc[0]['Norm'].split(",")

['Art 2 Abs 1 GG',
 '§\xa06 CoronaVGrundV TH',
 ' §\xa047 Abs 6 VwGO',
 ' §\xa032 IfSG',
 ' §\xa028 IfSG']

In [10]:
# Iterate over the rows of the dataframe
count = 0
for index, row in df_court.iterrows():
    if row['Norm'] != None or row['Norm'] != '' or row['Norm'] != ' ':
        count += 1

print(count)

249581


In [18]:
print(sum([sum(["§" in s for s in row['Norm'].split(",")]) if not row['Norm'] == None else 0 for index, row in df_court.iterrows()]))

2212277


In [20]:
2212277/len(df_court)

8.863964003670151

There are 2212277 legal paragraphs in the court decisions. On average there are 8.86 legal paragraphs tagged in a court decision